spec/steps/case_details/form_spec.rb

```diff
@@ -73,4 +75,26 @@
       expect(subject.main_offence).to eq('apples')
     end
   end
+
+  context 'when remitted to magistrate is provided' do
+    subject { described_class.new(arguments.merge(main_offence_suggestion:)) }
+
+    let(:remitted_to_magistrate_date) { Date.yesterday }
+    let(:remitted_to_magistrate) { 'yes' }
+
+    it 'magistrate date is provided' do
+      expect(form).to be_valid
+    end
+  end
+
+  context 'when remitted to magistrate date is not provided' do
+    subject { described_class.new(arguments.merge(main_offence_suggestion:)) }
+
+    let(:remitted_to_magistrate) { 'yes' }
+    let(:remitted_to_magistrate_date) { nil }
+
+    it 'magistrate date is not provided' do
+      expect(form).not_to be_valid
```

Comment:
can we add a test for the actual error and not just invalid (follow previous examples for this)